americanhandelsociety/americanhandelsociety-members
# [Issue 176] Add UI for publish_member_name_consent

## Description

This PR adds a UI for the new Member model field: `publish_member_name_consent`. It implements the following:

1. exposes the field in the admin dashboard (so the AHS newsletter editor can use it)
2. adds the field to the member-edit form
3. exposes the value of `publish_member_name_consent` (in a nice way) on the profile page

*N.b., The commit history for this PR is a little messy, since I had to revise my initial implementation after getting feedback from AHS stakeholders.*

## Manual testing

- [ ] create a super user

```
pipenv run python manage.py createsuperuser
```
- [ ]  use the admin view to assign yourself a "circle" tier
<img width="794" alt="Screenshot 2023-12-27 at 12 03 50 PM" src="https://github.com/americanhandelsociety/americanhandelsociety-members/assets/13078679/6e342c3b-b98f-4dac-a92e-63cb7e2f2a67">

- [ ] update your profile; select some value for the new "...do you consent to publishing your name...?" dropdown
- [ ] see your updated profile page

A non-circle user should NOT see the new field in the update form, nor should she see additional data on her profile page.
